diff --git a/src/crunch/crn.cpp b/src/crunch/crn.cpp
index 9d6e4fe..4127b2b 100644
--- a/src/crunch/crn.cpp
+++ b/src/crunch/crn.cpp
@@ -2,6 +2,8 @@
 #include <stddef.h> // For NULL, size_t
 #include <cstring> // for malloc etc
 
+#include <cstdint> // "C++11" uintptr_t definition
+
 #include "crn_decomp.h"
 
 #ifndef EXPORT
@@ -64,3 +66,8 @@ EXPORT void crn_decompress(void *src, unsigned int src_size, void *dst, unsigned
   crnd::crnd_unpack_level(pContext, pDecomp_images, dst_size, row_pitch, 0);
   crnd::crnd_unpack_end(pContext);
 }
+
+EXPORT void crn_set_memory_callbacks(void* pRealloc, void* pMsize, void* pUser_data) {
+  crnd::crnd_set_memory_callbacks(reinterpret_cast<crnd::crnd_realloc_func>(pRealloc), reinterpret_cast<crnd::crnd_msize_func>(pMsize), pUser_data);
+}
+
